pkg/controller/jobs/raycluster/raycluster_controller.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package raycluster
+
+import (
+	"context"
+	"strings"
+
+	rayclusterapi "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/podset"
+)
+
+var (
+	gvk = rayclusterapi.GroupVersion.WithKind("RayCluster")
+)
+
+const (
+	headGroupPodSetName = "head"
+	FrameworkName       = "ray.io/raycluster"
+)
+
+func init() {
+	utilruntime.Must(jobframework.RegisterIntegration(FrameworkName, jobframework.IntegrationCallbacks{
+		SetupIndexes:  SetupIndexes,
+		NewReconciler: NewReconciler,
+		SetupWebhook:  SetupRayClusterWebhook,
+		JobType:       &rayclusterapi.RayCluster{},
+		AddToScheme:   rayclusterapi.AddToScheme,
```

Comment:
Should `IsManagingObjectsOwner` be provided as well?

At the moment there is a "hack" in the Pod controller, inherited from the RayJob support:

https://github.com/kubernetes-sigs/kueue/blob/1579d75e5a35bdb4ab24c64b2da4e4effa98bbc8/pkg/controller/jobs/pod/pod_controller.go#L817-L822

That won't work for v1, and it may be better to take that opportunity to have that logic provided by the RayCluster integration itself.